tests/test_on_data/test_gin_ophys.py

```diff
@@ -148,6 +149,17 @@ class TestConverter(NWBConverter):
         nwb_segmentation = NwbSegmentationExtractor(file_path=nwbfile_path)
         check_segmentations_equal(segmentation, nwb_segmentation)
 
+    def test_extract_segmentation_interface_non_default_output_struct_name(self):
+        """Test that the value for 'output_struct_name' is propagated to the extractor level
+        where an error is raised."""
+        file_path = OPHYS_DATA_PATH / "segmentation_datasets" / "extract" / "extract_public_output.mat"
+        with self.assertRaisesWith(KeyError, '''"Unable to open object (object 'filters' doesn't exist)"'''):
```

Comment:
I feel like this is not the correct error that should be thrown by the extractor:
https://github.com/catalystneuro/roiextractors/blob/263b57598b91fc146632cdcb463485d8b33b0970/src/roiextractors/extractors/schnitzerextractor/extractsegmentationextractor.py#L82-L83
The current behavior is to redirect to the Legacy extractor when the struct name is not found in the file. what do you think @CodyCBakerPhD ?

Reply:
Ahhhh very good point.

What I would recommend is making https://github.com/catalystneuro/roiextractors/blob/master/src/roiextractors/extractors/schnitzerextractor/extractsegmentationextractor.py#L37

instead

```python
output_struct_name: Optional[str] = None,
```

and then split the version checking into two parts with different corresponding errors.

a) `if output_struct_name is not None` (which means it was specified by the user), then check if it is in the file. If not in the file, raise a specific error that the specified name is not in the file.

b) if it is `None`, then check if the name "output" is in the file, if it is try to check for the version info. If the name "output" is not in the file, inform user to specify the name of the struct in the file that has the info.

c) if the version field itself is not in the `self.output_struct_name` (with that name specified or not, but present either way) then we can assume at that point it is LEGACY

Reply:
Connecting PR https://github.com/catalystneuro/roiextractors/pull/210